### PR TITLE
refactor #1503: migrated kyc usecase to kotlin

### LINKED ISSUE #1503: refactor: migrate kyc usecase to kotlin

### PR DESCRIPTION
**Pull request by PratyushSingh07**
## Issue Fix
Fixes #1503 

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UpdateKYCLevel1Details.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    class ResponseValue : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues) {
+        mFineractRepository.updateKYCLevel1Details(
+            requestValues.clientId,
+            requestValues.kycLevel1Details
+        )
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeOn(Schedulers.io())
+            .subscribe(object : Subscriber<GenericResponse?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UpdateKYCLevel1Details.kt]: make it non-nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/FetchKYCLevel1Details.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Created by ankur on 24/May/2018
+ */
+class FetchKYCLevel1Details @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<FetchKYCLevel1Details.RequestValues, FetchKYCLevel1Details.ResponseValue>() {
+
+    class RequestValues(val clientId: Int) : UseCase.RequestValues
+    class ResponseValue(
+        val kycLevel1DetailsList: List<KYCLevel1Details?>
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/FetchKYCLevel1Details.kt:19]: make it non-nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/FetchKYCLevel1Details.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    ) : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues) {
+        mFineractRepository.fetchKYCLevel1Details(requestValues.clientId)
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeOn(Schedulers.io())
+            .subscribe(object : Subscriber<List<KYCLevel1Details?>?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/FetchKYCLevel1Details.kt]: make it non-nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/FetchKYCLevel1Details.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                override fun onCompleted() {}
+                override fun onError(e: Throwable) {
+                    useCaseCallback.onError(e.toString())
+                }
+
+                override fun onNext(kycLevel1Details: List<KYCLevel1Details?>?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/FetchKYCLevel1Details.kt]: make it non-nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UploadKYCDocs.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    class ResponseValue : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues) {
+        apiRepository.uploadKYCDocs(
+            requestValues.entitytype, requestValues.clientId,
+            requestValues.docname, requestValues.identityType, requestValues.file
+        )
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeOn(Schedulers.io())
+            .subscribe(object : Subscriber<GenericResponse?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UploadKYCDocs.kt]: make it non-nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UploadKYCDocs.kt [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                override fun onCompleted() {}
+                override fun onError(e: Throwable) {
+                    useCaseCallback.onError(e.toString())
+                }
+
+                override fun onNext(t: GenericResponse?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UploadKYCDocs.kt]: make it non-nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UploadKYCLevel1Details.kt [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    class ResponseValue : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues) {
+        mFineractRepository.uploadKYCLevel1Details(
+            requestValues.clientId,
+            requestValues.mKYCLevel1Details
+        )
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeOn(Schedulers.io())
+            .subscribe(object : Subscriber<GenericResponse?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UploadKYCLevel1Details.kt]: make it non-nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UploadKYCLevel1Details.kt [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+                override fun onCompleted() {}
+                override fun onError(e: Throwable) {
+                    useCaseCallback.onError(e.toString())
+                }
+
+                override fun onNext(t: GenericResponse?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/kyc/UploadKYCLevel1Details.kt]: make it non-nullable `?`

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/presenter/KYCLevel2Presenter.kt [hunk 9]
```diff
@@ -67,7 +67,7 @@ class KYCLevel2Presenter @Inject constructor(
         }
     }
 
-    override fun uploadKYCDocs(identityType: String?) {
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/presenter/KYCLevel2Presenter.kt:70]: Why removed `?`, have you checked the API? do expect identity type always

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/KYCContract.kt [hunk 10]
```diff
@@ -45,7 +45,7 @@ interface KYCContract {
         fun browseDocs()
         val context: Context
         fun updateFile(requestCode: Int, resultCode: Int, data: Intent?)
-        fun uploadKYCDocs(s: String?)
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/KYCContract.kt]: Why removed `?`, have you checked the API?

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Comment by PratyushSingh07** [comment]
done the changes . It is to be noted that certain endpoints especially the ones related to kyc, invoice, standing instructions etc arent working properly & i suspect this issue is backend related

**Pull request merged**